src/hrtim/external_event.rs

```diff
@@ -0,0 +1,371 @@
+use core::marker::PhantomData;
+
+use crate::comparator::{COMP1, COMP2, COMP3, COMP4, COMP5, COMP6, COMP7};
+use crate::gpio::gpiob::{PB3, PB4, PB5, PB6, PB7, PB8, PB9};
+use crate::gpio::gpioc::{PC11, PC12, PC5, PC6};
+use crate::gpio::{self, AF13, AF3};
+use crate::pwm::Polarity;
+use crate::stm32::HRTIM_COMMON;
+
+use super::control::HrTimCalibrated;
+
+#[derive(Copy, Clone, PartialEq)]
+pub struct ExternalEventSource<const N: u8, const IS_FAST: bool> {
+    _x: PhantomData<()>,
+}
+
+pub struct EevInputs {
+    pub eev_input1: EevInput<1>,
+    pub eev_input2: EevInput<2>,
+    pub eev_input3: EevInput<3>,
+    pub eev_input4: EevInput<4>,
+    pub eev_input5: EevInput<5>,
+    pub eev_input6: EevInput<6>,
+    pub eev_input7: EevInput<7>,
+    pub eev_input8: EevInput<8>,
+    pub eev_input9: EevInput<9>,
+    pub eev_input10: EevInput<10>,
+}
+
+impl EevInputs {
+    pub(crate) unsafe fn new() -> Self {
+        EevInputs {
+            eev_input1: EevInput { _x: PhantomData },
+            eev_input2: EevInput { _x: PhantomData },
+            eev_input3: EevInput { _x: PhantomData },
+            eev_input4: EevInput { _x: PhantomData },
+            eev_input5: EevInput { _x: PhantomData },
+            eev_input6: EevInput { _x: PhantomData },
+            eev_input7: EevInput { _x: PhantomData },
+            eev_input8: EevInput { _x: PhantomData },
+            eev_input9: EevInput { _x: PhantomData },
+            eev_input10: EevInput { _x: PhantomData },
+        }
+    }
+}
+
+pub struct EevInput<const N: u8> {
+    _x: PhantomData<()>,
+}
```

Comment:
With `#[non_exhaustive]`  you can make constructor private with using `PhantomData`.
```suggestion
impl EevInputs {
    pub(crate) unsafe fn new() -> Self {
        EevInputs {
            eev_input1: EevInput,
            eev_input2: EevInput,
            eev_input3: EevInput,
            eev_input4: EevInput,
            eev_input5: EevInput,
            eev_input6: EevInput,
            eev_input7: EevInput,
            eev_input8: EevInput,
            eev_input9: EevInput,
            eev_input10: EevInput,
        }
    }
}

#[non_exhaustive]
pub struct EevInput<const N: u8>;
```

Reply:
Wow, I had no idea. Thanks :)

Reply:
This was just example. There are several other empty `PhantomData<()>`.

Reply:
The others should be fixed now aswell